### PR TITLE
Responsiveness factfile entry

### PR DESCRIPTION
**Pull request by MitchellQ**
# This a pr is a responsive upgrade to the fact file details view

when a device is greater than 600 pixels and in landscape mode the view is rendered in a grid view

note there is a small issue with the height of this and will be corrected at a later stage. An git issue will be created for this

### also in this pr 
- the text widgets have been altered so that only the align property is named this reduces code duplication
- the body1 theme has been removed and added to the body text widget this allowed me to change the font size of widget when needed rather than applying the copy with method 
 - - note this should be considered for all themes 
- rotation on the home page has been added

### CODE REVIEW

**Comment by MitchellQ** [comment]
- Added a custom widget for the app bar on the activity views this has been tested on an older version of the application
- resolves issue #24

**Comment by MitchellQ** [comment]
Changed media queries to utilize the screen class math needs to be double checked where the percent param is passed in this resolves issue #23

**Comment by jacksos101** [comment]
FYI you have mixed up 'named constructors' with 'named parameters'. No issue.

**Comment by MitchellQ** [comment]
Which commit?

File: lib/views/activites/photograph_activity_view.dart [hunk 1]
```diff
@@ -104,13 +105,14 @@ class _PhotographActivityViewState extends State<PhotographActivityView> {
...
+                      "Your photo:",
                     ),
                     Padding(
                       padding: const EdgeInsets.only(top: 20.0),
                       child: Container(
-                        height: (MediaQuery.of(context).size.height / 100) * 45.2,
-                        width: (MediaQuery.of(context).size.width / 4) * 3.5,
+                        height:
+                            (MediaQuery.of(context).size.height / 100) * 45.2,
+                        width: Screen.width(context, percentage: 12.5),
```
Comment (jacksos101) [inline lib/views/activites/photograph_activity_view.dart]: Your math is wrong.

 1 / 4 * 3.5 = 87.5 - not 12.5

You'll need to go through and check all of your math.

File: lib/views/activites/photograph_activity_view.dart [hunk 2]
```diff
@@ -111,7 +111,7 @@ class _PhotographActivityViewState extends State<PhotographActivityView> {
                       padding: const EdgeInsets.only(top: 20.0),
                       child: Container(
                         height:
-                            (MediaQuery.of(context).size.height / 100) * 45.2,
+                            Screen.height(context, percentage: 45.2),
```
Comment (jacksos101) [inline lib/views/activites/photograph_activity_view.dart]: Can we make these round numbers. 45.2% is far to specific. Why not 45%?

File: lib/views/activites/picture_select_activity_view.dart [hunk 3]
```diff
@@ -84,8 +84,7 @@ class _PictureSelectActivityViewState extends State<PictureSelectActivityView> {
                     Column(
                       children: <Widget>[
                         Container(
-                          height: (MediaQuery.of(context).size.height / 100) *
```
Comment (jacksos101) [inline lib/views/activites/picture_select_activity_view.dart]: Please round

**Comment by jacksos101** [comment]
The one where you mention named constructors

Named constructor is like GridView.count().
Named parameter is like BodyText(text: 'fdafsdkjfa')

**Comment by jacksos101** [comment]
Alternatively, if you revert your media query replacements I will merge now

File: lib/views/activites/photograph_activity_view.dart (hunk 1 above)
Comment (MitchellQ) [reply lib/views/activites/photograph_activity_view.dart]: Will do i wasn't sure which way around these are supposed to be

**Comment by MitchellQ** [comment]
Can you revert them?

**Comment by jacksos101** [comment]
I'm not sure, I can do it using GitHub Desktop. But it might be less work to go through the values you have and just double check them? It looks like its just some of the width ones.

**Comment by jacksos101** [comment]
Doesn't have to be now

**Comment by MitchellQ** [comment]
Ok i will do it tomorrow then   what did you think of the app bar fix  it makes them all the same

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by jacksos101** [comment]
Haven't tested yet

File: lib/views/activites/photograph_activity_view.dart (hunk 2 above)
Comment (MitchellQ) [reply lib/views/activites/photograph_activity_view.dart]: When I converted them to be based off of width or height originally I tired to get them as close as possible to the original value it wont make much of a difference of small devices but on larger devices with larger widths it does make a difference so I think we should keep them unrounded maybe to 1 dp if we need to but I am happy with them as they are

File: lib/views/activites/picture_select_activity_view.dart (hunk 3 above)
Comment (MitchellQ) [reply lib/views/activites/picture_select_activity_view.dart]: I assume this is related to the above comment

**Pull request merged**